libraw/src/main/java/com/homesoft/photo/libraw/LibRaw.java

```diff
@@ -203,11 +192,26 @@ public Map<Float, float[]> getAvailableWhiteBalanceCoefficientsWithTemps() {
     public native int[][] getWhiteBalanceCoefficients();
     /**
      * Get a bitmap for the current image
-     * @return
+     * @deprecated Use {@link #getBitmap(Bitmap.Config)}
+     */
+    public Bitmap getBitmap() {
+        return getBitmap(Bitmap.Config.ARGB_8888);
+    }
+    /**
+     * Get a bitmap for the current image
+     * @deprecated Use {@link #getBitmap(Bitmap.Config)}
      */
-    public native Bitmap getBitmap();
     @RequiresApi(26)
-    public native Bitmap getBitmap16();
+    public Bitmap getBitmap16() {
+        return getBitmap(Bitmap.Config.RGBA_F16);
+    }
+    /**
+     * Return a mutable {@link Bitmap}.
+     * Requires that {@link #open(String)} or similar and {@link #dcrawProcess()} have been called.
+     * @param bitmapConfig Currently only {@link Bitmap.Config#ARGB_8888}
+     *                     and {@link Bitmap.Config#RGBA_F16}
+     */
+    public native Bitmap getMutableBitmap(Bitmap.Config bitmapConfig) throws IllegalArgumentException;
```

Comment:
Because of the way the bitmap is created it is always mutable.  In the C code I'm called Bitmap.createBitmap().  The code was modified in this PR to support ColorSpace, so it should be easy to find.  If you need an immutable version you can call Bitmap.copy().  I'm not sure I can create an immutable Bitmap because I need to mutate it to copy in the pixels from LibRaw.

FYI:  Hardware Bitmaps are technically mutable if the underlying HardwareBuffer is.

Reply:
ah good point, any bitmap created by `createBitmap` is mutable, and can be made immutable. this is great, thanks!